libs/packages/sam-formly/src/lib/formly-filters/sds-filters.component.ts

```diff
@@ -1,54 +1,64 @@
 import {
-  Component,
-  Input,
+  Component, Input, Output,
   ChangeDetectionStrategy,
-  Output,
-  EventEmitter,
-  OnChanges
+  EventEmitter, Optional, OnInit
 } from '@angular/core';
 import { FormGroup } from '@angular/forms';
 import { FormlyFieldConfig } from '@ngx-formly/core';
-import { Observable } from 'rxjs';
+import { Subject } from 'rxjs';
+import { SearchListComunicationService } from '@gsa-sam/components';
 
 @Component({
   selector: 'sds-filters',
   template: `
         <sds-accordion-item>
             <sds-accordion-item-header> {{accordionLabel}} </sds-accordion-item-header>
-            <formly-form [form]="form" [fields]="fields" [model]="model"></formly-form>
+            <formly-form [form]="form" (modelChange)="modelChange.next($event)" [fields]="fields" [model]="model"></formly-form>
         </sds-accordion-item>
     `,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class SdsFiltersComponent implements OnChanges {
+export class SdsFiltersComponent implements OnInit {
+
+  ngOnInit(): void {
+    this.modelChange.subscribe((change) => {
+      this.filterChange.emit(change);
+      if (this.searchListComunicationService) {
```

Comment:
why we want to depend on the searchListComunicationService this need to take care on implementation side..

Reply:
When you have sibling level components you should use a service to communicate between them. We have the sds-filters and the SearchListLayoutComponent  to communicate with each other in a standard way. This provides a way to have the implementation add this service as a provider and it takes care of that connection in a minimal way without the addition of copy and pasting of that interaction. This implementation also allows those components to be used independently still.

Reply:
This also still has the emit to be used outside of this constant

Reply:
Yes, you're correct this is for the communication, but this can be done on the implementation side, no need to added it to the component level.

Reply:
If it's already emiting then why we need that service in the component?

Reply:
I baked it in so that work without much effort on the implementation. Also as I said before that for sibling level communication of components which is what the sds-filters and the SearchListLayoutComponent are than it is recommended to use a service for that communication. Instead of passing the event up and right back down which is not recommended.

Reply:
This also goes back the email I sent Friday morning.

Reply:
I agree with you on creating service, but not in the component. The service can be added in the implementations side.

Reply:
I disagree. They will be used as sibling level components and that is what is recommended. The content being communicated is from the filter into the search list. To do that on the implementation side with the same design concept you would need to wrap both components which adds overhead we do not need. And that service is optional if those components was decided to work in another matter.

Reply:
If we're customizing the sds filter wrapper component to be used for results, I would suggest we move that out to the components library and rename it to be SdsResultsFilterComponent. Then it makes sense because it would always be paired with the other results components. In the current form, I don't think we want the generic formly filter component wrapper to have anything awareness of results (even optional).

Reply:
How about we move that service into the formly section with the filters and rename it to be more named with filters instead of search. Since our formly library may need to reference components later on (avoiding circular reference problem down the road).  Still gives the flexibility of both the parent child communication and that sibling level communication.